### ISSUE TITLE
Investigate use of Exhale for nicer docs

### ISSUE THREAD

**Issue by heinermann** [bwapi/bwapi#833]
https://exhale.readthedocs.io/en/latest/index.html

**Comment by ratiotile** [comment]
Can confirm that Sphinx (used by Exhale) is a good documentation generator.

**Comment by heinermann** [comment]
I've tried it and it just ... isn't quite there. It's missing a lot of doxygen commands and the list of functions, parameters, etc. are all in nested paragraph headings rather than properly formatted (i.e. notes aren't treated like notes, I'd have to rebuild the class/namespace lists from scratch to have it interact with the toc sidebar properly, and many other chores). It was too difficult to deal with. Don't get me wrong - it has a lot of capabilities and a lot of potential, it just doesn't take in the doxygen docs cleanly.

On the other hand I tried out the [m.css Doxygen wrapper/theme](https://mcss.mosra.cz/documentation/doxygen/) which preserves most things (with minor tweaks). The advantages over Doxygen plus my original custom scripts is that it does most of what I was doing in my scripts(minimizing structural layout, mobile friendly, html5, etc), has a working search, and is moderately pleasing to look at. Almost up to par out of the box.
